src/quotes_handler.rs

```diff
@@ -27,17 +68,62 @@ pub async fn identity(param: web::Query<Ident>) -> impl Responder {
             ))
             .await
     } else {
-        // place holder for identity quote code
-        HttpResponse::Ok()
-            .body(format!(
-                "Calling Identity Quote with nonce: {}",
-                param.nonce
-            ))
-            .await
+        let hash_alg =
+            get_hash_alg(config_get("cloud_agent", "tpm_hash_alg")?)?;
+
+        // generate new ephemeral rsa key
+        // TODO: or are we supposed to get this from somewhere?
+        // TODO: where to keep priv key? we'll need it later for U, V
+        let (nk_pub, _nk_priv) = rsa_generate_pair(2048)?;
+        let nk_digest = pubkey_to_tpm_digest(&nk_pub, hash_alg)?;
+
+        // lock TPM context and regenerate AIK handle
+        let mut context = tpmcontext.lock().unwrap(); //#[allow_ci] (see https://github.com/rust-lang-nursery/failure/issues/192)
+        let (ek_handle, _, _) = create_ek(&mut context, Some(Rsa))?;
+        let (ak_handle, _, _) = create_ak(&mut context, ek_handle)?;
+
+        // TODO: does the pcr need to be reset before it is extended? (seems like no?)
+        // ex: https://github.com/keylime/keylime/blob/master/keylime/tpm/tpm_main.py#L953
+
+        // extend NK into PCR16
+        // TODO: is PCR16 the right place?
+        context.pcr_extend(PcrHandle::Pcr16, nk_digest);
+
+        // reader's note: this hashing algorithm further specifies the PCR;
+        // it is NOT re-hashing the contents of the PCR.
+        let pcrlist = PcrSelectionListBuilder::new()
+            .with_selection(hash_alg, &[PcrSlot::Slot16])
+            .build();
+
+        let sig_scheme = get_sig_scheme(TpmSigScheme::default())?;
+        let nonce = param.nonce.to_owned().into_bytes().try_into().map_err(|err| Error::Tpm {
+            err,
+            kind: None,
+            message: format!("Could not convert nonce {:? }into tss_esapi::structures::Data", param.nonce)
+        } )?;
+
+        let (attestation, sig) = context
+            .quote(ak_handle, &nonce, sig_scheme, pcrlist)
+            .map_err(|err| Error::Tpm {
+                err,
+                kind: None,
+                message: format!("TPM identity quote error: {:?}", err),
+            })?;
+
+        let idquote = IdentQuote {
+            quote: "placeholderquote".as_bytes(),
+            signature: "placeholdersig".as_bytes(),
+            pcrblob: "placeholderpcrblob".as_bytes(),
+            nk_pub: &nk_pub.public_key_to_pem().unwrap(), //#[allow_ci] can't convert openssl ErrorStack to ResponseError
```

Comment:
I'm OK with this as is since it's an iteration but could the `unwrap` be replaced with a `match` before this block with a rewrap of the error?